### ISSUE TITLE
Validate database names

### ISSUE THREAD

**Issue by AlekSi** [FerretDB/FerretDB#807]
After #750.

MongoDB database names restrictions are documented there: https://www.mongodb.com/docs/manual/reference/limits/#naming-restrictions
PostgreSQL restrictions – there: https://www.postgresql.org/docs/current/sql-syntax-lexical.html

For collection names, we use a special settings table in the schema matching database name. Let's not complicate things for database/schema names. Instead, let's limit database names to what PostgreSQL supports.

Let's validate that database/schema name:
* matches regular expression `^[a-z_][a-z0-9_]{0,62}$`;
* can't start with a reserved prefix `_ferretdb_` (just in case).

We could relax those rules in the future.

* [ ] implement database name validation on database/schema creation;
* [ ] add integration test;
* [ ] document the difference in rules in README.md;
* [ ] add diff dance test.

Tigris is not in scope.

**Comment by AlekSi** [comment]
@seeforschauer Please don't forget about the diff test in dance

**Comment by seeforschauer** [comment]
@AlekSi yeah, I just remember that the dance test fails w/o FerretDB changes in main.

**Comment by seeforschauer** [comment]
dance tests left.
